packages/core/src/components/DataTable/index.tsx

```diff
@@ -128,7 +138,28 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
     (...args) => JSON.stringify(args),
   );
 
-  componentDidUpdate(prevProps: DataTableProps) {
+  componentDidUpdate(prevProps: DataTableProps, prevState: State) {
+    const { data, filterData } = this.props;
+    const { sortBy, sortDirection, selectedRows } = this.state;
+    const sortedData: IndexedParentRow[] = this.getData(data!, sortBy, sortDirection, selectedRows);
+    const filteredData = filterData!(sortedData);
+    const oldFilteredData = prevProps.filterData!(sortedData);
+
+    if (
+      filteredData.length > 0 &&
+      (filteredData.length !== oldFilteredData.length ||
+        filteredData.some(
+          (x: IndexedParentRow, i: number) =>
+            x.metadata.originalIndex !== oldFilteredData[i].metadata.originalIndex,
+        ))
+    ) {
+      // We need to make sure the cache is cleared before React tries to re-render.
+      setTimeout(() => {
```

Comment:
I guess technically this only needs to be called if `dynamicRowHeight === true`?

Reply:
yeah - add a check for `dynamicRowHeight` as the first condition in the `if` statement